providers/apis/defi/raydium/price_fetcher.go

```diff
@@ -0,0 +1,256 @@
+package raydium
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	binary "github.com/gagliardetto/binary"
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/programs/token"
+	"github.com/gagliardetto/solana-go/rpc"
+	"go.uber.org/zap"
+
+	"github.com/skip-mev/slinky/oracle/config"
+	oracletypes "github.com/skip-mev/slinky/oracle/types"
+	"github.com/skip-mev/slinky/pkg/math"
+	oraclemath "github.com/skip-mev/slinky/pkg/math/oracle"
+	providertypes "github.com/skip-mev/slinky/providers/types"
+	mmtypes "github.com/skip-mev/slinky/x/marketmap/types"
+)
+
+var _ oracletypes.PriceAPIFetcher = (*APIPriceFetcher)(nil)
+
+// SolanaJSONRPCClient is the expected interface for a solana JSON-RPC client according
+// to the APIPriceFetcher.
+//
+//go:generate mockery --name SolanaJSONRPCClient --output ./mocks/ --case underscore
+type SolanaJSONRPCClient interface {
```

Comment:
nit: I would probably put this interface in another package if it's supposed to be general across any solana defi provider we support.

Reply:
That makes sense, I feel like we can do that when we implement the next provider? Or are you requesting a change to the dir structure of the `defi` directory?